test/mocks/MockTokenClient.ts

```diff
@@ -8,40 +8,28 @@ export class MockTokenClient extends TokenClient {
     [chainId: number]: { [token: string]: { deposits: number[]; totalRequirement: BigNumber } };
   } = {};
 
-  setTokenData(chainId: number, token: string, balance: BigNumber, allowance: BigNumber = toBN(0)) {
+  setTokenData(chainId: number, token: string, balance: BigNumber, allowance: BigNumber = toBN(0)): void {
     if (!this.tokenData[chainId]) {
       this.tokenData[chainId] = {};
     }
     this.tokenData[chainId][token] = { balance, allowance };
   }
-  setTokenShortFallData(chainId: number, token: string, deposits: number[], totalRequirement: BigNumber) {
+  setTokenShortFallData(chainId: number, token: string, deposits: number[], totalRequirement: BigNumber): void {
     if (!this.tokenShortfall[chainId]) {
       this.tokenShortfall[chainId] = {};
     }
     this.tokenShortfall[chainId][token] = { deposits, totalRequirement };
   }
 
-  getBalance(chainId: number, token: string) {
-    if (!this.tokenData[chainId]) {
-      return toBN(0);
-    }
-    if (!this.tokenData[chainId][token]) {
-      return toBN(0);
-    }
-    return this.tokenData[chainId][token].balance;
+  getBalance(chainId: number, token: string): BigNumber {
+    return this.tokenData[chainId]?.[token]?.balance || toBN(0);
```

Comment:
Does the || have any impact here? For numeric values it'd activate if balance was 0, but this seems to be a BigNumber object.

```suggestion
    return this.tokenData[chainId]?.[token]?.balance ?? toBN(0);
```

Reply:
The `||` here will convert even if the `balance` is falsy (potentially 0). I think there should be no functional issue with this.

Reply:
Agree - no functional issue since || will catch a superset of ??. Was just curious if there was a specific motivation for it.